proteus/NumericalFlux.py

```diff
@@ -1798,11 +1798,25 @@ def __init__(self,vt,getPointwiseBoundaryConditions,
         self.hasInterior=False
     def setDirichletValues(self,ebqe):
         for ci in range(self.nc):
-            self.ebqe[('u',ci)].flat[:] = ebqe[('u',ci)].flat[:]
-            for (ebNE,k),g,x in zip(list(self.DOFBoundaryConditionsDictList[ci].keys()),
-                                    list(self.DOFBoundaryConditionsDictList[ci].values()),
-                                    list(self.DOFBoundaryPointDictList[ci].values())):
-                self.ebqe[('u',ci)][ebNE,k]=g(x,self.vt.timeIntegration.t)
+            try:
```

Comment:
@adimako here is an example of how to add a try/except statement to allow for the g(x,t,n) function interface for Dirichlet conditions.  I tested this on a simple duct problem. For Kappa, etc., you may need to make a change like this to the setDirichletValues member of some other NumericalFlux classes. https://github.com/erdc/proteus/blob/master/proteus/mprans/Kappa.py#L69

Reply:
I only need the normals for the Dirichlet conditions, so I won't change advective / diffusive. Placeholders will be there in case someone wants to use it for other models